Movies-MVC/Movies-MVC/Controllers/HomeViewController.swift

```diff
@@ -0,0 +1,263 @@
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.backgroundView = UIImageView(image: UIImage(named: Constants.backGroundView))
+        tableView.register(MovieTableViewCell.self, forCellReuseIdentifier: Constants.movieIdentifier)
+        tableView.showsVerticalScrollIndicator = false
+        tableView.separatorStyle = .none
+        view.addSubview(tableView)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            tableView.topAnchor.constraint(equalTo: topRatedButton.bottomAnchor, constant: 10),
+            tableView.leftAnchor.constraint(equalTo: view.leftAnchor),
+            tableView.rightAnchor.constraint(equalTo: view.rightAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+    }
+
+    private func setPadding() {
+        currentRow += 20
+        tableView.reloadData()
+    }
+}
+
+// MARK: - UITableViewDataSource
+
+extension HomeViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        movies.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.movieIdentifier, for: indexPath)
+            as? MovieTableViewCell else { return UITableViewCell() }
+        let movie = movies[indexPath.row]
+        cell.backgroundColor = .clear
+        cell.setDescription(model: movie)
+        return cell
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let detailMovie = movies[indexPath.row]
+        let detailVC = DetailViewController()
+        detailVC.imageName = detailMovie.poster
+        networkManager.fetchCreditsResult(detailMovie.id) { movie in
+            DispatchQueue.main.async {
+                detailVC.title = detailMovie.title
+                guard let movieDetail = try? movie.get().cast else { return }
+                detailVC.id = String(detailMovie.id)
+                detailVC.movieDescription = detailMovie.overview
+                detailVC.actorNames = movieDetail.map { $0.name ?? "" }
+                detailVC.actorImageNames = movieDetail.map { $0.profilePath ?? "" }
+                detailVC.rating = detailMovie.voteAverage
+                detailVC.relize = detailMovie.releaseDate
+                self.navigationController?.pushViewController(detailVC, animated: true)
+            }
+        }
+    }
+
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        if indexPath.row == currentRow + 18 {
+            switch title {
+            case Constants.MovieTypes.popular.rawValue:
+                currentPage += 1
+                networkManager.fetchMoviesResult(Url.populary, currentPage) { movies in
```

Comment:
```suggestion
                networkManager.fetchMoviesResult(Url.populary, currentPage) { movies in
```
в методы надо вынести